src/natten/flex.py

```diff
@@ -141,8 +146,14 @@ def flex_na1d(
             f"got {query.dtype=}, {key.dtype=}, {value.dtype=}."
         )
 
-    batch_size, seqlen, num_heads, head_dim = query.shape
-    input_size = (seqlen,)
+    batch_size, seq_length, num_heads, head_dim = query.shape
+    input_size = (seq_length,)
+
+    if not can_run_flex_attention(query.shape):
+        raise ValueError(f"FlexAttention does only supports sequence lengths "
+                         f"divisible by 128, and head dimension being a power "
+                         f"of 2 and at most 256. Got {seq_length=} and "
+                         f"{head_dim=}.")
```

Comment:
Better if we can separate the two errors, but I guess that way we repeat `can_run_flex_attention`. Although we could just have a "fail mode", but nvm, it's fine the way it is.

Just make these corrections to the error message, and the ones below.
```suggestion
        raise ValueError(f"FlexAttention only supports sequence lengths "
                         f"divisible by 128, and head dim must be a power "
                         f"of 2. Got {seq_length=} and {head_dim=}.")
```